File: Sorting/Bubble_Sort/Bubble_Sort.py


def bubble_sort(arr):
    n = len(arr)
    for i in range(0, n - 1):
        for k in range(0, n - i - 1):
            if arr[k] > arr[k+1]:
                tmp = arr[k]
                arr[k] = arr[k-1]
                arr[k-1] = tmp

File: Sorting/Bubble_Sort/Bubble_Sort.c
#include <stdio.h>

void swap(int *m, int *n){
    int temp = *m;
    *m = *n;
    *n = temp;
}

void bubble_sort(int * arr){
    int i, j;
    int n = sizeof(arr)/sizeof(arr[0]);
    for (i = 0; i < n - 1; i++){
        for (j = 0; j < n - i - 1; j++){
            if (arr[j] > arr[j+1]){
                swap((arr+j), (arr+j+1));
            }else;
        }
    }
}